src/System.Data.SqlClient/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -1071,5 +1071,73 @@ static internal Version GetAssemblyVersion()
 
             return s_systemDataVersion;
         }
+
+
+        static internal readonly string[] AzureSqlServerEndpoints = {Res.GetString(Res.AZURESQL_GenericEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_GermanEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_UsGovEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_ChinaEndpoint) };
+
+        static internal bool IsAzureSqlServerEndpoint(string dataSource)
+        {
+            Debug.Assert(!string.IsNullOrWhiteSpace(dataSource));
+
+            string serverName = ExtractServerName(dataSource);
```

Comment:
Do we really need to extract the whhole server name? Seems like string.EndsWith() would handle most of the logic. The only cases you'd need to account for are port or instance name at the end.

Named pipes aren't supported in Azure connections, so you can just autofail that case.